### ISSUE TITLE
Feature request: support ensuring exactly one newline at EoF

### ISSUE THREAD

**Issue by byronbowerman** [jsmecham/nova-whitespace#1]
Just what it says on the tin.

I took a look at the [`enureSingleTrailingNewline`][0] feature in atom's whitespace package, but it uses an API that doesn't seem available in Nova (getting editor contents by line).

Is something like this feasible with the current Nova extension api?

[0]: https://github.com/atom/whitespace/blob/0099ad18813e3b9ace047347f3c957118f19839a/lib/whitespace.js#L190-L207

**Comment by jsmecham** [comment]
This is actually already on my todo list. Thanks for the suggestion!

**Comment by edwardloveall** [comment]
I started to look into adding this, but I discovered when editing this extension nova was inserting the new line for me. Here's a gif of that happening:

![newline-on-save](https://user-images.githubusercontent.com/49549/99191317-f1636280-2763-11eb-8fb9-eca8491c0af6.gif)

I'm not sure what's going on here. I thought it might be the `.editorconfig` settings but I copied them into a new empty project and it didn't add the new line. I'm not sure how this is working. 😅
